account_multicurrency_revaluation/model/res_currency.py

```diff
@@ -12,19 +12,18 @@ class ResCurrency(models.Model):
     _inherit = 'res.currency'
 
     @api.model
-    def _get_conversion_rate(self, from_currency, to_currency):
+    def _get_conversion_rate(self, from_currency, to_currency, company, date):
         context = self.env.context
-        if 'revaluation' in context:
-            rate = from_currency.rate
-            if rate == 0.0:
-                date = context.get('date', time.strftime('%Y-%m-%d'))
-                raise UserError(
-                    _('No rate found '
-                      'for the currency: %s '
-                      'at the date: %s') %
-                    (from_currency.symbol, date)
-                )
-            return 1.0 / rate
-
-        else:
-            return super()._get_conversion_rate(from_currency, to_currency)
+        if not context.get('revaluation'):
+            return super()._get_conversion_rate(from_currency, to_currency,
+                                                company, date)
+        rate = from_currency.rate
+        if rate == 0.0:
+            date = context.get('date', time.strftime('%Y-%m-%d'))
+            raise UserError(
+                _('No rate found '
+                  'for the currency: %s '
+                  'at the date: %s') %
+                (from_currency.symbol, date)
+            )
```

Comment:
@goliveirab 
So far this condition is never met as `_get_conversion_rate` [[1](https://github.com/odoo/odoo/blob/12.0/odoo/addons/base/models/res_currency.py#L183-L186)] calls `_get_rates` [[2](https://github.com/odoo/odoo/blob/12.0/odoo/addons/base/models/res_currency.py#L48-L59)]  and it always COALESCEs to 1 if none rate is found, so risk of zero division is none.


```
        if rate == 0.0:
            date = context.get('date', time.strftime('%Y-%m-%d'))
            raise UserError(
                _('No rate found '
                  'for the currency: %s '
                  'at the date: %s') %
                (from_currency.symbol, date)
            )
```

Get rid of it.

cc/ @moylop260 @fclementic2c